### PR TITLE
[chore] awsproxy extension ReportFatalError -> ReportStatus

### PR DESCRIPTION
**Pull request by atoulme**
Remove use of deprecated host.ReportFatalError

Linked to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30501

### CODE REVIEW

File: extension/awsproxy/extension.go
Comment (codeboten) [inline extension/awsproxy/extension.go:19]: is this needed anymore with settings being part of the struct?

Comment (atoulme) [reply extension/awsproxy/extension.go:19]: not necessarily but I applied minimal changes, codeowners may choose to tidy afterwards or ask for more changes.

**Comment by atoulme** [comment]
@Aneurysm9 are the changes in this PR acceptable?

**Comment by atoulme** [comment]
I will bring this up at the SIG meeting tomorrow, see if we can unblock review.

**Review by bryan-aguilar — APPROVED** [review]

**Review by Aneurysm9 — APPROVED** [review]

**Pull request merged**
